.github/workflows/test-and-publish.yaml

```diff
@@ -47,7 +50,10 @@ jobs:
         run: ./setup.sh #&& ./build_script.sh
       - name: Build wheel
         run: |
+          source $VENV # activate virtualenv
+          echo $VIRTUAL_ENV
           echo $(poetry run python --version)
+          rm -rf ./build/
```

Comment:
Got the correct Python version with the cached `./build` deleted.
It seems falling back to the cached values in `build/CMakeCache.txt`.
We shouldn't cache `./build` dir anyhow.

Reply:
I guess it's because the virtualenv installed by poetry doesn't have the `Development` components as in `find_package(Python 3.9...<3.12 COMPONENTS Interpreter Development REQUIRED)`

Reply:
Ouhh that's a weird symptom.... Why did this not happen with my branch?